pkg/datapath/linux/ethtool/ethtool_linux.go

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 Authors of Cilium
+
+package ethtool
+
+import (
+	"bytes"
+
+	"golang.org/x/sys/unix"
+)
+
+func IsVirtualDriver(iface string) (bool, error) {
+	fd, err := unix.Socket(unix.AF_INET, unix.SOCK_DGRAM, unix.IPPROTO_IP)
+	if err != nil {
+		return false, err
+	}
+	defer unix.Close(fd)
+
+	info, err := unix.IoctlGetEthtoolDrvinfo(fd, iface)
+	if err != nil {
+		return false, err
+	}
+	return string(bytes.TrimRight(info.Driver[:], "\x00")) == "veth", nil
```

Comment:
Is this only ever called on veth ifs? Shouldn't this include bridges and mac/ipvlan as well? If not, the fn name might be a bit misleading.

```
# ethtool -i eth1
driver: macvlan
...
```

Reply:
Currently, the only call site is here:

https://github.com/cilium/cilium/blob/2d075ea34720ed625e297ede67ace8dce5bbdb53/pkg/datapath/loader/base.go#L182

used in ENI IPAM mode top determine which interfaces to use for IPSec encryption. I'm not very familiar with that, so I'd have to defer to @jrfastab to answer the question of whether we should include bridges and mac/ipvlan as well.